### PR TITLE
Update/link in bio flow

### PR DESCRIPTION
**Pull request by renancarvalho**
## Proposed Changes

Organize the steps order for the whole Link in bio flow. Some steps are fake until they get complete.

## Testing Instructions

- Pull this branch and `yarn start` from the root folder
- Start at the beginning of the flow http://calypso.localhost:3000/setup/intro?flow=link-in-bio
- The flow order should be: ( according to the latest design )

1. Intro / Welcome 
2. If logged out - > account creation;   If not Choose a design (fake step for now)
3. Link in bio setup
4. Choose a domain 
5. Choose a plan
6. Pre checkout;  `fake processing step for now`
7. Checkout 
8. Pre launchpad `fake for now`
9. Launchpad  

  - test with logged in and logged out
  - test purchase and free plan

image

Related to #66576

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const LinkInBioSetup: Step = function LinkInBioSetup( { navigation } ) {
 				// communicate the error to the user
 			}
 		}
-		submit?.( { siteTitle, tagline } );
+		goNext();
```
Comment (arcangelini) [inline client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx]: Should this be checking for `siteTitle` first and then be submitting it?

File: client/signup/config/flows-pure.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ export function generateFlows( {
 	getDestinationFromIntent = noop,
 	getDIFMSignupDestination = noop,
 	getDIFMSiteContentCollectionDestination = noop,
-	getStepperFlowDestination = noop,
+	// getStepperFlowDestination = noop,
```
Comment (arcangelini) [inline client/signup/config/flows-pure.js]: You can delete this along with the definition in `client/signup/config/flows.js`. This was used before we changed up the order and only existed for these two steps which are now gone.

**Comment by renancarvalho** [comment]
Some styling issue found during testing:


- [x] 		WP logo jumps slightly between intro and “choose design” steps in link-in-bio flows, when logged in, and when on big screen/Chrome.
- [x] 		Setup page: continue button doesn’t have hover/active states
- [x] 		On design picking step, arrows felt little too invisible and no hover/click states
- [x] 		On design picking step, lack of scroll, hover or cursor indicators felt odd when clicking blurred designs.
- [x] 		On mobile “Powered by JP” doesn’t have bottom margin at intro page ( Could not reproduce it )
- [ ] 		At domain page I saw “accept cookies” banner all of a sudden again. ( will be checked in another PR )
- [ ] 		Was missing drag and drop support on avatar button, but not a blocker. ( will be checked in another PR ))

**Review by arcangelini — COMMENTED** [review]
This looks good so far, it does look like we are missing the styling for the `/start` processing step to match the figma. As of right now it flashes the unstyled loading screen rather than the new styled version. You can ignore this comment if we are planning on addressing this in another PR 😃

File: client/signup/config/flows-pure.js [hunk 3]
```diff
@@ -129,14 +129,11 @@ export function generateFlows( {
...
 			name: 'link-in-bio',
-			steps: getAddOnsStep(
-				isEnabled( 'signup/professional-email-step' )
-					? [ 'user', 'domains', 'emails', 'plans' ]
-					: [ 'user', 'domains', 'plans' ]
-			),
-			destination: ( dependencies ) => getStepperFlowDestination( dependencies, 'link-in-bio' ),
+			steps: [ 'domains', 'plans' ],
+			destination: ( dependencies ) =>
+				`/setup/completingPurchase?flow=link-in-bio&siteSlug=${ dependencies.siteSlug }`,
```
Comment (alshakero) [inline client/signup/config/flows-pure.js]:
```suggestion
				`/setup/completingPurchase?flow=link-in-bio&siteSlug=${ encodeURIComponent( dependencies.siteSlug ) }`,
```

File: client/signup/config/flows-pure.js (hunk 2 above)
Comment (renancarvalho) [reply client/signup/config/flows-pure.js]: Done! Thank you for this tip @arcangelini

**Review by alshakero — CHANGES_REQUESTED** [review]
Love it. 

- ✅ Setting site title
- ✅ Setting site description
- ✅ Setting site logo
- ✅ Logged in and out
- ✅ Free and paid plans
- ⚠️ The "Preparing Checkout" phrase appears even when the user picks free plan and domain, this might scare some users. Can we check if they have free stuff before showing it?

**Review by alshakero — APPROVED** [review]
We'll deal with the "Preparing checkout" issue in a follow up.

File: client/signup/config/flows-pure.js (hunk 3 above)
Comment (renancarvalho) [reply client/signup/config/flows-pure.js]: Thank you for your review @alshakero, the 'preparing checkout' issue you pointed will be handled on the next PR.
🥇

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7491913

Thank you @renancarvalho for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7491913).

**Pull request merged**